File: Week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);

int main(void)
{
    string text = get_string("Text: ");
    printf("letters: %i \n", count_letters(text));
    printf("words: %i \n", count_words(text));
}




int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if ((text[i] >= 97 && text[i] <= 122) || (text[i] >= 65 && text[i] <= 90))
        {
            letters ++;
        }
    }

    return letters;
}

int count_words(string text)
{
    int words = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 32)
        {
            words++;
        }
    }

    return words;
}

File: Week 6/sentimental-readability/readability.py
# TODO

from cs50 import get_string
import re


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = (letters / words) * 100
    S = (sentences / words) * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def count_letters(text):
    letters = 0

    for i in text:
        if i.isalpha():
            letters += 1

    return letters


def count_words(text):
    words = text.split()

    return len(words)


def count_sentences(text):
    sentences = re.split(r'[.!?]+', text)

    return len(sentences) - 1


if __name__ == '__main__':
    # This code won't run if this file is imported.
    main()